lib/devicetrust/native/device_windows.go

```diff
@@ -20,51 +20,64 @@ import (
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/base64"
+	"fmt"
 	"math/big"
 	"os"
 	"os/exec"
 	"os/user"
 	"path/filepath"
+	"time"
 
 	"github.com/google/go-attestation/attest"
 	"github.com/gravitational/trace"
 	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
 	"github.com/gravitational/teleport/lib/devicetrust"
+	"github.com/gravitational/teleport/lib/windowsexec"
 )
 
 const (
-	deviceStateFolderName  = ".teleport-device"
-	attestationKeyFileName = "attestation.key"
+	deviceStateFolderName        = ".teleport-device"
+	attestationKeyFileName       = "attestation.key"
+	credentialActivationFileName = "credential-activation"
 )
 
+type deviceState struct {
+	attestationKeyPath       string
+	credentialActivationPath string
+}
+
 // setupDeviceStateDir ensures that device state directory exists.
 // It returns the absolute path to where the attestation key can be found:
 // $CONFIG_DIR/teleport-device/attestation.key
-func setupDeviceStateDir(getBaseDir func() (string, error)) (akPath string, err error) {
+func setupDeviceStateDir(getBaseDir func() (string, error)) (*deviceState, error) {
 	base, err := getBaseDir()
 	if err != nil {
-		return "", trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
 
 	deviceStateDirPath := filepath.Join(base, deviceStateFolderName)
-	keyPath := filepath.Join(deviceStateDirPath, attestationKeyFileName)
+	ds := &deviceState{
+		attestationKeyPath:       filepath.Join(deviceStateDirPath, attestationKeyFileName),
+		credentialActivationPath: filepath.Join(deviceStateDirPath, credentialActivationFileName),
+	}
 
 	if _, err := os.Stat(deviceStateDirPath); err != nil {
 		if os.IsNotExist(err) {
 			// If it doesn't exist, we can create it and return as we know
 			// the perms are correct as we created it.
 			if err := os.Mkdir(deviceStateDirPath, 700); err != nil {
-				return "", trace.Wrap(err)
+				return nil, trace.Wrap(err)
 			}
-			return keyPath, nil
+			return ds, nil
 		}
-		return "", trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
```

Comment:
Handle only failures here, otherwise let it flow to the end of the function?

```go
	switch _, err := os.Stat(deviceStateDirPath); {
	case os.IsNotExist(err):
		// If it doesn't exist, we can create it and return as we know
		// the perms are correct as we created it.
		if err := os.Mkdir(deviceStateDirPath, 700); err != nil {
			return nil, trace.Wrap(err)
		}
	case err != nil:
		return nil, trace.Wrap(err)
	}
```

Reply:
819c6596af